tools/SourceKit/tools/sourcekitd-test/CMakeLists.txt

```diff
@@ -18,9 +18,14 @@ else()
   target_link_libraries(sourcekitd-test PRIVATE sourcekitd)
 endif()
 if(NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+  set(arch_subdir "${SWIFTLIB_DIR}/${SWIFT_SDK_${SWIFT_HOST_VARIANT_SDK}_LIB_SUBDIR}/${SWIFT_HOST_VARIANT_ARCH}")
+  target_link_directories(sourcekitd-test PRIVATE "${arch_subdir}")
   target_link_libraries(sourcekitd-test PRIVATE
     dispatch
-    BlocksRuntime)
+    BlocksRuntime
+    "${arch_subdir}/swiftrt${CMAKE_C_OUTPUT_EXTENSION}"
```

Comment:
how did this work before? 😨

Reply:
That is an excellent question, and honestly, I'm not sure. It quite clearly needs `swiftrt.o` though, so…

Reply:
Hmmm. Not sure this is the right fix, looking at it a bit more. There's an `AddSwiftSourceKit.cmake` file that I think _should_ already be adding these files, which is presumably how it worked before. Obviously that's going wrong somehow now.

Reply:
Ohhhh… I bet it's linking the wrong `swiftrt.o`.

Reply:
Trying to reproduce the problem locally. In the meantime I'm going to get rid of this change, and re-run CI so I can see the errors again.

Reply:
If we held CMake correctly, it would be using swiftc as the linker here instead of clang, which would then pick up swiftrt.o and the stdlib.

Reply:
Indeed, instead of the complicated dance in `AddSwiftSourceKit.cmake` (which doesn't look entirely right to me; at the very least, I'm suspicious because it does nothing on Windows, which seems… wrong).